snow/engine/snowman/engine_test.go

```diff
@@ -3182,3 +3185,35 @@ func TestShouldIssueBlock(t *testing.T) {
 		})
 	}
 }
+
+type mockConnVDR struct {
+	tracker.Peers
+	percent float64
+}
+
+func (m *mockConnVDR) ConnectedPercent() float64 {
+	return m.percent
+}
+
+func TestEngineAbortQueryWhenInPartition(t *testing.T) {
+	require := require.New(t)
+
+	// Buffer to record the log entries
+	buff := gbytes.NewBuffer()
+
+	conf := DefaultConfig(t)
+	// Overwrite the log to record what it says
+	conf.Ctx.Log = logging.NewLogger("", logging.NewWrappedCore(logging.Info, buff, logging.Plain.ConsoleEncoder()))
+	conf.Params = snowball.DefaultParameters
+	conf.ConnectedValidators = &mockConnVDR{percent: 0.7, Peers: conf.ConnectedValidators}
+
+	_, _, _, _, engine := setup(t, conf)
+
+	// Gossip will cause a pull query if enough stake is connected
+	require.NoError(engine.Gossip(context.Background()))
+
+	shouldSay := gbytes.Say("WARN snowman/engine.go:935 dropped query for block {\"reason\": \"insufficient stake connected\", \"blkID\": \"j7sgHpXKn6y22DQdBNr5bzidt345dor9L8rL3Mxo9bt91k4ta\", \"ratio\": 0.7}")
```

Comment:
If we move the line where this log is emitted from, will it cause this test to fail?

Maybe we could match the log for just the error string?

Another alternative would be to use the `TestSender` and fail the test if `Gossip` triggers a query unexpectedly.

Reply:
> If we move the line where this log is emitted from, will it cause this test to fail?

Yeah good point. I was in a hurry (it was really late here :-) ) so I just copy-pasted everything.